src/back/hlsl/writer.rs

```diff
@@ -311,15 +823,79 @@ impl<W: Write> Writer<W> {
     /// Doesn't add any newlines or leading/trailing spaces
     fn write_constant(&mut self, module: &Module, handle: Handle<Constant>) -> BackendResult {
         let constant = &module.constants[handle];
-        #[allow(clippy::match_single_binding)]
         match constant.inner {
+            crate::ConstantInner::Scalar {
+                width: _,
+                ref value,
+            } => {
+                self.write_scalar_value(*value)?;
+            }
             _ => {
                 return Err(Error::Unimplemented(format!(
                     "write_constant {:?}",
                     constant
                 )))
             }
         }
+
+        Ok(())
+    }
+
+    /// Helper method used to write [`ScalarValue`](ScalarValue)
+    ///
+    /// # Notes
+    /// Adds no trailing or leading whitespace
+    fn write_scalar_value(&mut self, value: ScalarValue) -> BackendResult {
+        match value {
+            ScalarValue::Sint(value) => write!(self.out, "{}", value)?,
+            ScalarValue::Uint(value) => write!(self.out, "{}u", value)?,
+            // Floats are written using `Debug` instead of `Display` because it always appends the
```

Comment:
pretty cool!

Reply:
It's a copy-paste from `glsl-out` :) There are a lot of copy-paste between at least 3 backends: glsl, wgsl, hlsl 🤔